scripts/eqtl_hail_batch/conditional_analysis.py

```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env python3
+
+"""Perform conditional analysis on SNPs and expression residuals"""
+
+import os
+import hail as hl
+import hailtop.batch as hb
+import pandas as pd
+import statsmodels.api as sm
+from patsy import dmatrices  # pylint: disable=no-name-in-module
+from scipy.stats import spearmanr
+from cpg_utils.hail_batch import (
+    dataset_path,
+    output_path,
+    copy_common_env,
+    init_batch,
+    remote_tmpdir,
+)
+from cloudpathlib import AnyPath
+import click
+from multipy.fdr import qvalue
+
+
+DEFAULT_DRIVER_MEMORY = '4G'
+DRIVER_IMAGE = 'australia-southeast1-docker.pkg.dev/cpg-common/images/multipy:0.16'
+assert DRIVER_IMAGE
+
+TOB_WGS = dataset_path('mt/v7.mt/')
+FREQ_TABLE = dataset_path('joint-calling/v7/variant_qc/frequencies.ht/', 'analysis')
+
+
+def get_number_of_scatters(residual_df: pd.DataFrame, significant_snps_df: pd.DataFrame) -> int:
+    """get index of total number of genes"""
+
+    # Identify the top eSNP for each eGene and assign remaining to df
```

Comment:
(Nitpick): you don't actually assign the remaining SNPs to the df in this function/chunk of code (I think?)

Reply:
Good catch! I removed that bit